### PR TITLE
Update of english translation.

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: docs/en/development/style.md [hunk 1]
```diff
@@ -2,816 +2,831 @@
...
 
-   ```cpp
-    inline void readBoolText(bool & x, ReadBuffer & buf)
-    {
-        char tmp = '0';
-        readChar(tmp, buf);
-        x = tmp != '0';
-    }
-   ```
+**2.** Offsets are 4 spaces. Configure your development environment so that a tab adds four spaces.
```
Comment (alexey-milovidov) [inline docs/en/development/style.md]: Bad translation. It should be "indentation", not "offsets".

File: docs/en/index.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ We'll say that the following is true for the OLAP (online analytical processing)
 - Data is updated in fairly large batches (> 1000 rows), not by single rows; or it is not updated at all.
 - Data is added to the DB but is not modified.
 - For reads, quite a large number of rows are extracted from the DB, but only a small subset of columns.
-- Tables are "wide", meaning they contain a large number of columns.
+- Tables are "wide," meaning they contain a large number of columns.
```
Comment (alexey-milovidov) [inline docs/en/index.md]: Bad merge.

File: docs/en/interfaces/http_interface.md [hunk 3]
```diff
@@ -130,14 +130,13 @@ POST 'http://localhost:8123/?query=DROP TABLE t'
 
 For successful requests that don't return a data table, an empty response body is returned.
 
-You can use compression when transmitting data.
+You can use the internal ClickHouse compression format when transmitting data. The compressed data has a non-standard format, and you will need to use the special clickhouse-compressor program to work with it (it is installed with the clickhouse-client package).
```
Comment (alexey-milovidov) [inline docs/en/interfaces/http_interface.md]: Bad merge.

File: docs/en/system_tables/system.functions.md [hunk 4]
```diff
@@ -6,6 +6,6 @@ Columns:
 
 ```text
 name String          – Function name.
-is_aggregate UInt8   – Whether it is an aggregate function.
+is_aggregate UInt8    – Whether it is an aggregate function.
```
Comment (alexey-milovidov) [inline docs/en/system_tables/system.functions.md]: ?

File: docs/ru/table_engines/mysql.md [hunk 5]
```diff
@@ -2,14 +2,14 @@
 
 # MySQL
 
-Движок MySQL позволяет выполнять SELECT запросы над данными, хранящимися на удалённом MySQL сервере.
+Движок MySQL позволяет выполнять `SELECT` запросы над данными, хранящимися на удалённом MySQL сервере.
 
-Движок принимает 4 параметра: адрес сервера (хост и порт); имя базы данных; имя таблицы; имя пользоваля; пароль пользователя. Пример:
+Формат вызова:
```
Comment (alexey-milovidov) [inline docs/ru/table_engines/mysql.md]: ?

File: docs/en/index.md (hunk 2 above)
Comment (BayoNet) [reply docs/en/index.md]: By typographic rules, commas should be inside the quotation marks. https://h.yandex-team.ru/?https%3A%2F%2Fdocs.microsoft.com%2Fen-us%2Fstyle-guide%2Fpunctuation%2Fquotation-marks

File: docs/en/development/style.md (hunk 1 above)
Comment (BayoNet) [reply docs/en/development/style.md]: I'll fix it

File: docs/en/interfaces/http_interface.md (hunk 3 above)
Comment (BayoNet) [reply docs/en/interfaces/http_interface.md]: There is no bad merge. English version is exactly the same as russian one in this place.

File: docs/en/system_tables/system.functions.md (hunk 4 above)
Comment (BayoNet) [reply docs/en/system_tables/system.functions.md]: I'll reformat it at all.

File: docs/ru/table_engines/mysql.md (hunk 5 above)
Comment (BayoNet) [reply docs/ru/table_engines/mysql.md]: I have decided that:
"
Формат вызова:

```
MySQL('host:port', 'database', 'table', 'user', 'password');
```
"
enough to not need additional text.

File: docs/ru/query_language/queries.md [hunk 6]
```diff
@@ -308,10 +308,10 @@ SELECT * FROM system.parts WHERE active
...
 total 48
-drwxrwxrwx 2 clickhouse clickhouse 20480 мая   13 02:58 20140317_20140323_2_2_0
-drwxrwxrwx 2 clickhouse clickhouse 20480 мая   13 02:58 20140317_20140323_4_4_0
-drwxrwxrwx 2 clickhouse clickhouse  4096 мая   13 02:55 detached
--rw-rw-rw- 1 clickhouse clickhouse     2 мая   13 02:58 increment.txt
+drwxrwxrwx 2 clickhouse clickhouse 20480 may   13 02:58 20140317_20140323_2_2_0
+drwxrwxrwx 2 clickhouse clickhouse 20480 may   13 02:58 20140317_20140323_4_4_0
+drwxrwxrwx 2 clickhouse clickhouse  4096 may   13 02:55 detached
+-rw-rw-rw- 1 clickhouse clickhouse     2 may   13 02:58 increment.txt
 ```
```
Comment (filimonov) [inline docs/ru/query_language/queries.md]: Enlish bash types month with first capital letter, and without extra spaces after. i.e. `May 13` not `may    13`.

File: docs/en/getting_started/example_datasets/wikistat.md [hunk 7]
```diff
@@ -20,8 +20,5 @@ CREATE TABLE wikistat
...
-ls -1 /opt/wikistat/ | grep gz | while read i; do echo $i; gzip -cd /opt/wikistat/$i | ./wikistat-loader --time="$(echo -n $i | sed -r 's/pagecounts-([0-9]{4})([0-9]{2})([0-9]{2})-([0-9]{2})([0-9]{2})([0-9]{2})\.gz/\1-\2-\3 \4-00-00/')" | clickhouse-client --query="INSERT INTO wikistat FORMAT TabSeparated"; done
+for i in {2007..2016}; do for j in {01..12}; do echo $i-$j >&2; curl -sSL "http://dumps.wikimedia.org/other/pagecounts-raw/$i/$i-$j/" | grep -oE 'pagecounts-[0-9]+-[0-9]+\.gz'; done; done | sort | uniq | tee links.txtcat links.txt | while read link; do wget http://dumps.wikimedia.org/other/pagecounts-raw/$(echo $link | sed -r 's/pagecounts-([0-9]{4})([0-9]{2})[0-9]{2}-[0-9]+\.gz/\1/')/$(echo $link | sed -r 's/pagecounts-([0-9]{4})([0-9]{2})[0-9]{2}-[0-9]+\.gz/\1-\2/')/$link; donels -1 /opt/wikistat/ | grep gz | while read i; do echo $i; gzip -cd /opt/wikistat/$i | ./wikistat-loader --time="$(echo -n $i | sed -r 's/pagecounts-([0-9]{4})([0-9]{2})([0-9]{2})-([0-9]{2})([0-9]{2})([0-9]{2})\.gz/\1-\2-\3 \4-00-00/')" | clickhouse-client --query="INSERT INTO wikistat FORMAT TabSeparated"; done
```
[the remark this answers is not in the document]
Comment (BayoNet) [reply docs/en/getting_started/example_datasets/wikistat.md]: I'll fix it

File: docs/en/functions/other_functions.md [hunk 8]
```diff
@@ -278,4 +277,3 @@ The inverse function of MACNumToString. If the MAC address has an invalid format
 ## MACStringToOUI(s)
 
 Accepts a MAC address in the format AA:BB:CC:DD:EE:FF (colon-separated numbers in hexadecimal form). Returns the first three octets as a UInt64 number. If the MAC address has an invalid format, it returns 0.
-
```
[the remark this answers is not in the document]
Comment (BayoNet) [reply docs/en/functions/other_functions.md]: Really I don't know where that chimpanzees from, but I'll feed (or fix) them.

File: docs/en/functions/other_functions.md [hunk 9]
```diff
@@ -59,8 +59,7 @@ For elements in a nested data structure, the function checks for the existence o
 
 Allows building a unicode-art diagram.
 
-`bar (x, min, max, width)` – Draws a band with a width proportional to (x - min) and equal to 'width' characters when x == max.
-`min, max` – Integer constants. The value must fit in Int64.`width` – Constant, positive number, may be a fraction.
+`bar (x, min, max, width)` – Draws a band with a width proportional to (x - min) and equal to 'width' characters when x == max.`min, max` – Integer constants. The value must fit in Int64.`width` – Constant, positive number, may be a fraction.
```
[the remark this answers is not in the document]
Comment (BayoNet) [reply docs/en/functions/other_functions.md]: No, wrong formatting. In russian version too. I'll fix.

File: docs/en/operations/settings/query_complexity.md [hunk 10]
```diff
@@ -26,21 +26,40 @@ After enabling readonly mode, you can't disable it in the current session.
...
+## max_memory_usage_for_all_queries
+
+The maximum amount of RAM to use for running all queries on a single server.
 
-- Large constants (for example, a very long string constant).
-- The states of certain aggregate functions.
+Default values are defined in [Settings.h](https://github.com/yandex/ClickHouse/blob/master/dbms/src/Interpreters/Settings.h#L245). By default, the amount is not restricted (`max_memory_usage_for_all_queries = 0`).
 
-Memory consumption is not fully considered for aggregate function states 'min', 'max', 'any', 'anyLast', 'argMin', and 'argMax' from String and Array arguments.
+See also the descriptions of [max_memory_usage]( and #settings_max_memory_usage).
```
[the remark this answers is not in the document]
Comment (BayoNet) [reply docs/en/operations/settings/query_complexity.md]: Wrong translation way. I'll fix.

File: docs/en/operations/settings/settings_profiles.md [hunk 11]
```diff
@@ -15,13 +15,9 @@ Example:
 ```xml
 <!-- Settings profiles -->
 <profiles>
-    <!-- Default settings -->
+    <!Default settings -->
```
[the remark this answers is not in the document]
Comment (BayoNet) [reply docs/en/operations/settings/settings_profiles.md]: I'll fix.

File: docs/en/table_engines/index.md [hunk 12]
```diff
@@ -8,7 +8,8 @@ The table engine (type of table) determines:
 - Use of indexes, if present.
 - Whether multithreaded request execution is possible.
 - Data replication.
-- When reading data, the engine is only required to extract the necessary set of columns.However, in some cases, the query may be partially processed inside the table engine.
+- When reading data, the engine is only required to extract the necessary set of columns.
+        However, in some cases, the query may be partially processed inside the table engine.
```
[the remark this answers is not in the document]
Comment (BayoNet) [reply docs/en/table_engines/index.md]: Wrong initial formatting in russian sources of documentation.

File: docs/en/table_engines/merge.md [hunk 13]
```diff
@@ -2,21 +2,21 @@
 
 The Merge engine (not to be confused with `MergeTree`) does not store data itself, but allows reading from any number of other tables simultaneously.
 Reading is automatically parallelized. Writing to a table is not supported. When reading, the indexes of tables that are actually being read are used, if they exist.
-The Merge engine accepts parameters: the database name and a regular expression for tables. Example:
+The Merge engine accepts parameters: the database name and a regular expression for tables. Example.
```
[the remark this answers is not in the document]
Comment (BayoNet) [reply docs/en/table_engines/merge.md]: Wrong syntax in russian sources. I'll fix.

File: docs/ru/query_language/queries.md (hunk 6 above)
Comment (BayoNet) [reply docs/ru/query_language/queries.md]: My fault. I'll fix.

File: docs/en/dicts/external_dicts_dict_layout.md [hunk 14]
```diff
@@ -2,11 +2,11 @@
...
+There are a [variety of ways](#dicts-external_dicts_dict_layout-manner) to store dictionaries in memory.
 
-We recommend [flat](external_dicts_dict_layout#dicts-external_dicts_dict_layout-flat), [hashed](external_dicts_dict_layout#dicts-external_dicts_dict_layout-hashed), and [complex_key_hashed](external_dicts_dict_layout#dicts-external_dicts_dict_layout-complex_key_hashed). which provide optimal processing speed.
+We recommend [flat](#dicts-external_dicts_dict_layout-flat), [hashed](#dicts-external_dicts_dict_layout-hashed)and[complex_key_hashed](#dicts-external_dicts_dict_layout-complex_key_hashed). which provide optimal processing speed.
 
-Caching is not recommended because of potentially poor performance and difficulties in selecting optimal parameters. Read more about this in the "[cache](external_dicts_dict_layout#dicts-external_dicts_dict_layout-cache)" section.
+Caching is not recommended because of potentially poor performance and difficulties in selecting optimal parameters. Read more in the section " [cache](#dicts-external_dicts_dict_layout-cache)".
```
Comment (filimonov) [inline docs/en/dicts/external_dicts_dict_layout.md]: extra space after quote

File: docs/en/functions/string_search_functions.md [hunk 15]
```diff
@@ -5,14 +5,16 @@ The search substring or regular expression must be a constant in all these funct
 
 ## position(haystack, needle)
 
-Search for the 'needle' substring in the 'haystack' string.
+Search for the `needle` substring in the `haystack` string.
 Returns the position (in bytes) of the found substring, starting from 1, or returns 0 if the substring was not found.
-It has also chimpanzees.
```
Comment (filimonov) [inline docs/en/functions/string_search_functions.md]: I will miss them :)

File: docs/en/table_engines/dictionary.md [hunk 16]
```diff
@@ -102,5 +103,5 @@ LIMIT 1
 │        152689 │ Некоторый товар │
```
Comment (filimonov) [inline docs/en/table_engines/dictionary.md]: Russian sample

File: docs/ru/dicts/external_dicts_dict_layout.md [hunk 17]
```diff
@@ -2,11 +2,11 @@
 
 # Хранение словарей в памяти
 
-Словари можно размещать в памяти [множеством способов](external_dicts_dict_layout#dicts-external_dicts_dict_layout-manner).
+Словари можно размещать в памяти [множеством способов](#dicts-external_dicts_dict_layout-manner).
 
-Рекомендуем [flat](external_dicts_dict_layout#dicts-external_dicts_dict_layout-flat), [hashed](external_dicts_dict_layout#dicts-external_dicts_dict_layout-hashed) и [complex_key_hashed](external_dicts_dict_layout#dicts-external_dicts_dict_layout-complex_key_hashed). Скорость обработки словарей при этом максимальна.
+Рекомендуем [flat](#dicts-external_dicts_dict_layout-flat), [hashed](#dicts-external_dicts_dict_layout-hashed) и [complex_key_hashed](#dicts-external_dicts_dict_layout-complex_key_hashed). Скорость обработки словарей при этом максимальна.
 
-Размещение с кэшированием не рекомендуется использовать из-за потенциально низкой производительности и сложностей в подборе оптимальных параметров. Читайте об этом подробнее в разделе " [cache](external_dicts_dict_layout#dicts-external_dicts_dict_layout-cache)".
```
Comment (filimonov) [inline docs/ru/dicts/external_dicts_dict_layout.md]: лишний пробел после кавычки.

**Review by filimonov — COMMENTED** [review]
:+1:

**Pull request merged**